SimTracker/TrackTriggerAssociation/interface/TTClusterAssociationMap.h

```diff
@@ -296,13 +297,12 @@ bool TTClusterAssociationMap<T>::isCombinatoric(
 }
 
 template <typename T>
-edm::Ptr<TrackingParticle> TTClusterAssociationMap<T>::findTrackingParticlePtr(
-    edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const {
+TrackingParticlePtr TTClusterAssociationMap<T>::findTrackingParticlePtr(TTClusterRefT<T> aCluster) const {
   if (this->isGenuine(aCluster)) {
     return this->findTrackingParticlePtrs(aCluster).at(0);
   }
 
-  edm::Ptr<TrackingParticle>* temp = new edm::Ptr<TrackingParticle>();
+  TrackingParticlePtr* temp = new TrackingParticlePtr();
```

Comment:
potential memory leak

Reply:
Looks more like a guaranteed memory leak to me (nothing deletes `temp`). This and the following line could be replaced with
```cpp
return TrackingParticlePtr();
```

Reply:
done